datafusion/core/tests/sql/set_variable.rs

```diff
@@ -260,37 +260,238 @@ async fn set_u64_variable_bad_value() {
+        "| name                           | setting |",
+        "+--------------------------------+---------+",
+        "| datafusion.execution.time_zone | +08:00  |",
+        "+--------------------------------+---------+",
+    ];
+    assert_batches_eq!(expected, &result);
+}
 
-    assert_eq!(
-        err.to_string(),
-        "Error during planning: Changing Time Zone isn't supported yet"
-    );
+#[tokio::test]
+async fn set_time_zone_with_alias_variable_name() {
+    let ctx =
+        SessionContext::with_config(SessionConfig::new().with_information_schema(true));
 
-    // for alias time zone
-    let err = plan_and_collect(&ctx, "set time zone = '8'")
+    // TIME ZONE with space
+    plan_and_collect(&ctx, "SET TIME ZONE = '+08:00'")
         .await
-        .unwrap_err();
+        .unwrap();
 
-    assert_eq!(
-        err.to_string(),
-        "Error during planning: Changing Time Zone isn't supported yet"
-    );
+    let result = plan_and_collect(&ctx, "SHOW TIME ZONE").await.unwrap();
+    let expected = vec![
+        "+--------------------------------+---------+",
+        "| name                           | setting |",
+        "+--------------------------------+---------+",
+        "| datafusion.execution.time_zone | +08:00  |",
+        "+--------------------------------+---------+",
+    ];
+    assert_batches_eq!(expected, &result);
 
-    // for alias timezone
-    let err = plan_and_collect(&ctx, "set timezone = '8'")
+    // TIMEZONE without space
+    plan_and_collect(&ctx, "SET TIMEZONE = '+07:00'")
         .await
-        .unwrap_err();
+        .unwrap();
 
-    assert_eq!(
-        err.to_string(),
-        "Error during planning: Changing Time Zone isn't supported yet"
-    );
+    let result = plan_and_collect(&ctx, "SHOW TIMEZONE").await.unwrap();
+    let expected = vec![
+        "+--------------------------------+---------+",
+        "| name                           | setting |",
+        "+--------------------------------+---------+",
+        "| datafusion.execution.time_zone | +07:00  |",
+        "+--------------------------------+---------+",
+    ];
+    assert_batches_eq!(expected, &result);
+}
+
+#[tokio::test]
+async fn set_time_zone_good_time_zone_format() {
+    let ctx =
+        SessionContext::with_config(SessionConfig::new().with_information_schema(true));
+
+    plan_and_collect(&ctx, "SET TIME ZONE = '+08:00'")
+        .await
+        .unwrap();
+
+    // casting UTF-8 to TimestampTZ isn't supported yet, add Timestamp as the middle layer for now
+    let result =
+        plan_and_collect(&ctx, "SELECT '2000-01-01T00:00:00'::TIMESTAMP::TIMESTAMPTZ")
+            .await
+            .unwrap();
+    let expected = vec![
+        "+-----------------------------+",
+        "| Utf8(\"2000-01-01T00:00:00\") |",
+        "+-----------------------------+",
+        "| 2000-01-01T08:00:00+08:00   |",
+        "+-----------------------------+",
+    ];
```

Comment:
ideally we could change this to `'2000-01-01T00:00:00'::TIMESTAMPTZ` once it's added in the casting kernel (in arrow-rs)